sphinx/builders/linkcheck.py

```diff
@@ -225,9 +228,15 @@ def check() -> Tuple[str, str, int]:
 
     def process_result(self, result: Tuple[str, str, int, str, str, int]) -> None:
         uri, docname, lineno, status, info, code = result
+
+        linkstat = dict(docname=docname, lineno=lineno,
+                        status=status, code=code, uri=uri, info=info)
+        linkstat['docpath'] = docpath = self.env.doc2path(docname, None)
```

Comment:
I think defining `docpath` (or filename) before creating linkstat will make this code simple.
```
docpath = self.env.doc2path(...)
linkstat = dict(..., docpath=docpath)
```

Reply:
done.